docs/DeveloperGuide.adoc

```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
+Th event feature is largely centered around the `Event` class, and is integrated into ConTAct in a similar fashion to the address book.
+The main commands supporting this feature is `schedule` and `cancel`, which both interact with the `Calendar` model.
+The `undo/redo` and `clear` commands were also modified to extend towards the calendar model.
+
+The Event class has these fields that the user can specify to schedule an event:
 
 * Event Name: The name of the event, valid if it is not an empty string.
 * Date: The date of the event, valid if expressed in this forms:  `DD-MM-YYYY`.
 It must also be a date that actually falls within a calendar and between the years 1600 to 9999 (i.e. dates such as 25-13-2000 are invalid).
 * Start Time: The start time of the event, valid in this form: `HH:MM`, in 24-hour format.
 * End Time: The end time of the event, valid if it is in this form: `HH:MM`, in 24-hour format.
 The end time must also come after the start time.
-* Description: The description of the event, valid if it is not an empty string.
+* Description: (OPTIONAL) The description of the event, valid if it is not an empty string.
+
+During runtime, the events are stored in an observable list in the model architecture.
+This list is exposed to the UI to display the list accordingly.
+Furthermore, the event data is permanently stored in a `calendar.xml` file for use between sessions.
+
+Below are several commands that allow the user to interact with the events:
 
-Currently, the main operation is the `schedule` command, and it is implemented as such:
+===== Schedule
 
-* `schedule event/EVENT_NAME date/DATE start/START_TIME end/END_TIME descr/DESCRIPTION`
+The `schedule` command allows the user to add a unique event to the calendar, and it is implemented as such:
 
-This operation exists within the logic component of ConTAct, with the Event data structure falling
-under the models component.
+* `schedule event/EVENT_NAME date/DATE start/START_TIME end/END_TIME [descr/DESCRIPTION]`
 
 Below is an example of how the schedule command behaves:
 
-`schedule event/Office Christmas Party date/23-12-2018 start/18:00 end/20:00 descr/Party at Scranton Branch`
+`schedule event/CS2103 Tutorial 1 date/23-01-2018 start/15:00 end/16:00 descr/Introduction to Software Engineering`
 
-This will simply create an event with the specified fields and store it similar to how the addressbook stores a student.
-Furthermore, the schedule command ensures that all the fields specified must be valid for it to be a success.
+This will simply create an event with the specified fields and store it accordingly.
 
-Once scheduled, the list of events will be sorted by chronological order (taking the date and the start time).
+Once scheduled, the events will be inserted in its sorted position (by chronological order).
 
 The GUI itself will have a calendar component which will display the user's current list of events that they have
 scheduled, while adjusting for the current date.
```

Comment:
what does "adjusting for the current date" mean? does it automatically scroll to show the first upcoming event